ddtrace/contrib/pytest/plugin.py

```diff
@@ -83,6 +84,11 @@ def pytest_configure(config):
         Pin(tags=ci_tags, _config=ddtrace.config.pytest).onto(config)
 
 
+def pytest_sessionstart(session):
+    pin = Pin.get_from(session.config)
+    pin.tracer._span_processors.append(SpanSetOriginTagProcessor(origin_tag=ci.CI_APP_TEST_ORIGIN))
```

Comment:
note: we want to be sure we don't append this processor more than once

it can also get cleared/reset if the tracer is recreated or on fork

Reply:
isn't it reset on `_initialize_span_processors` in `ddtrace/tracer.py` ? Do we reset span/trace processors anywhere else? So I can have a look at how it's done

Reply:
The problem is likely from `Tracer._child_after_fork`

https://github.com/DataDog/dd-trace-py/blob/23827ed051c2890bc831c0bfde89e06f2792d60f/ddtrace/tracer.py#L431-L442

If there is any forking that occurs during a test run (or use of multiprocessing, etc), then it'll get wiped out.

Reply:
I wonder if we just have a `CIAppSpanProcessor` that is always configured, but then a config flag to toggle whether it does anything or not... ?

I still don't think that is an elegant approach though.